src/models/state/wallet/wallet_state.rs

```diff
@@ -3734,4 +3900,196 @@ mod tests {
+            let data_dir = unit_test_data_directory(network).unwrap();
+
+            // generate events
+            let genesis_block = Block::genesis(network);
+            let premine_receiver = mock_genesis_global_state(
+                network,
+                0,
+                WalletEntropy::devnet_wallet(),
+                cli_args::Args::default(),
+            )
+            .await;
+            let premine_change_key = premine_receiver
+                .lock_guard()
+                .await
+                .wallet_state
+                .nth_spending_key(KeyType::Symmetric, 0)
+                .expect("wallet should be capable of generating symmetric spending keys");
+            let now =
+                genesis_block.header().timestamp + Timestamp::months(6) + Timestamp::minutes(5);
+            let sender_randomness = rng.random();
+            let alice_future_spending_key = alice_secret.nth_generation_spending_key(20);
+            let tx_output = TxOutput::onchain_native_currency(
+                NativeCurrencyAmount::coins(1),
+                sender_randomness,
+                alice_future_spending_key.to_address().into(),
+                false,
+            );
+            let (transaction, _, _) = premine_receiver
+                .lock_guard()
+                .await
+                .create_transaction_with_prover_capability(
+                    vec![tx_output].into(),
+                    premine_change_key,
+                    UtxoNotificationMedium::OffChain,
+                    NativeCurrencyAmount::coins(0),
+                    now,
+                    TxProvingCapability::PrimitiveWitness,
+                    &JobQueue::dummy(),
+                )
+                .await
+                .unwrap();
+            let block_1 = invalid_block_with_transaction(&genesis_block, transaction);
+
+            // some possible CLI configurations:
+            // scan mode is inactive
+            let cli_default = cli_args::Args::default();
+            // scan mode is active but looking in the wrong blocks
+            let cli_wrong_range = cli_args::Args {
+                scan_blocks: Some(10..=100),
+                ..Default::default()
+            };
+            // scan mode is active but scanning for too few keys
+            let cli_too_few_keys = cli_args::Args {
+                scan_keys: Some(5),
+                ..Default::default()
+            };
+            // scan mode is active and scanning for the right keys in the right
+            // blocks
+            let cli_well_configured = cli_args::Args {
+                scan_blocks: Some(0..=u64::MAX),
+                scan_keys: Some(25),
+                ..Default::default()
+            };
+
+            for (case, cli_args, should_catch_utxo) in [
+                (0, cli_default, false),
+                (1, cli_wrong_range, false),
+                (2, cli_too_few_keys, false),
+                (3, cli_well_configured, true),
```

Comment:
Why not `.enumerate()`?

Reply:
These were used for debugging but don't serve a purpose anymore. Removing.